core/store/models/common.go

```diff
@@ -457,10 +483,28 @@ func (d *Duration) UnmarshalJSON(input []byte) error {
 	if err != nil {
 		return err
 	}
-	*d = Duration(v)
+	*d, err = MakeDuration(v)
+	if err != nil {
+		return err
+	}
 	return nil
 }
 
+func (d *Duration) Scan(v interface{}) (err error) {
```

Comment:
Postgres actually has a built-in duration type (called `interval`).

We might consider using that instead.

Reply:
I'm reluctant because that increases our dependency on postgres. What would the benefits be?

Reply:
Thanks, I created story #172460686 for that.